nidirect_cold_weather_payments/src/Form/ColdWeatherPaymentCheckerForm.php

```diff
@@ -184,40 +200,56 @@ public function cwpCheck(array $form, FormStateInterface $form_state) {
    */
   public function validateForm(array &$form, FormStateInterface $form_state) {
     parent::validateForm($form, $form_state);
+    // Validation is handled by ajax callback - or if no JS, by submitForm().
+  }
 
-    // Adding this validation to take care of older browsers.
-    $postcode = $form_state->getValue('postcode');
-
-    if (!empty($postcode)) {
-      $postcode_district = $this->cwpGetPostcodeDistrict($postcode);
-
-      if (!is_numeric($postcode_district) || $postcode_district < 1 || $postcode_district > 99) {
-        $form_state->setErrorByName('postcode', $this->t('Postcode must be a valid Northern Ireland postcode.'));
-      }
+  /**
+   * Validates that a postcode is a valid NI postcode or outward code (first half of the postcode).
+   */
+  protected function validatePostcode(array &$form, FormStateInterface $form_state) {
```

Comment:
I'd be a bit more specific with the function name otherwise it might be mistaken to validate any UK postcode. Maybe consider something like isValidNIPostcode()